src/components/dataset/settings.vue

```diff
@@ -0,0 +1,133 @@
+-->
+<template>
+  <div id="dataset-settings">
+    <div v-if="dataset.dataExists" class="panel panel-simple">
+      <div class="panel-heading">
+        <h1 class="panel-title">{{ $t('entityWorkflow') }}</h1>
+      </div>
+      <div class="panel-body">
+        <form id="dataset-settings-form">
+          <div class="radio">
+            <label>
+              <input v-model="approvalRequired" name="approvalRequired" type="radio" :value="false"
+                aria-describedby="dataset-setting-on-receipt" :aria-disabled="dataset.awaitingResponse" @change="update()">
+              <strong>{{ $t('onReceipt.label') }}</strong>
+            </label>
+            <p id="dataset-setting-on-receipt" class="help-block">
+              {{ $t('onReceipt.description') }}
+            </p>
+          </div>
+          <div class="radio">
+            <label>
+              <input v-model="approvalRequired" name="approvalRequired" type="radio" :value="true"
+                aria-describedby="dataset-setting-on-approval" :aria-disabled="dataset.awaitingResponse" @change="update()">
+              <strong>{{ $t('onApproval.label') }}</strong>
+            </label>
+            <p id="dataset-setting-on-approval" class="help-block">
+              {{ $t('onApproval.description') }}
+            </p>
+          </div>
+        </form>
+      </div>
+    </div>
+
+    <dataset-pending-submissions v-bind="pendingSubmissionModal" @hide="hideAndReset" @success="hideAndUpdate"/>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'DatasetSettings'
+};
+</script>
+
+<script setup>
+import { ref, watch, inject } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+import DatasetPendingSubmissions from './pending-submissions.vue';
+
+import { apiPaths, isProblem } from '../../util/request';
+import { noop } from '../../util/util';
+import { useRequestData } from '../../request-data';
+
+const { t } = useI18n();
+
+const alert = inject('alert');
+
+const { project, dataset } = useRequestData();
+
+const approvalRequired = ref(dataset.approvalRequired);
+
+const pendingSubmissionModal = ref({
+  state: false,
+  pendingSubmissions: 0
+});
+
+watch(() => dataset.dataExists, () => {
+  approvalRequired.value = dataset.approvalRequired;
+});
+
+const update = (convert) => {
+  if (dataset.awaitingResponse) return;
+
+  dataset.request({
+    method: 'PATCH',
+    url: apiPaths.dataset(project.id, dataset.name, { convert }),
+    data: { approvalRequired: approvalRequired.value },
+    fulfillProblem: ({ code }) => code === 400.29,
+    patch: ({ data }) => {
+      if (isProblem(data)) {
+        pendingSubmissionModal.value.pendingSubmissions = data.details.count;
+        pendingSubmissionModal.value.state = true;
+      } else {
+        dataset.approvalRequired = data.approvalRequired;
+        alert.success(dataset.approvalRequired ? t('onApproval.successMessage') : t('onReceipt.successMessage'));
+      }
+    }
+  })
+    .catch(noop);
```

Comment:
Should we reset `approvalRequired.value` to `dataset.approvalRequired` if there's an error? Otherwise, to retry the request, the user would first have to change the radio back to its previous state, which would send a request.